panos_upgrade_assurance/check_firewall.py

```diff
@@ -870,6 +871,38 @@ def check_ssl_cert_requirements(self, rsa: dict = {}, ecdsa: dict = {}) -> Check
         result.status = CheckStatus.SUCCESS
         return result
 
+    def check_non_finished_jobs(self) -> CheckResult:
+        """Check for any job with status different than FIN.
+
+        # Returns
+
+        CheckResult: Object of [`CheckResult`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkresult) class taking \
+            value of:
+
+        * [`CheckStatus.SUCCESS`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) when all jobs are in FIN state.
+        * [`CheckStatus.FAIL`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) otherwise, `CheckResult.reason`
+            field contains information about the 1<sup>st</sup> job found with status different than FIN (job ID and the actual
+            status).
+        * [`CheckStatus.SKIPPED`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) when there are no jobs on a
+            device.
+
+        """
+        result = CheckResult()
+
+        all_jobs = self._node.get_jobs()
+
+        if all_jobs:
+            for jid, job in all_jobs.items():
+                if job["status"] != "FIN":
+                    result.reason = f"At least one job (ID={jid}) is not in finished state (state={job['status']})."
```

Comment:
Shall we write "status" in the parentheses which shows the value from API response?

```suggestion
                    result.reason = f"At least one job (ID={jid}) is not in finished state (status={job['status']})."
```

Reply:
then it wouldn't be consistent, `...finished state (status=...)`, so state or status? :)

Reply:
Yes that's also the case, it just looks like an API response field to me, I guess because it's in parentheses with an equal sign. That might be me though :)